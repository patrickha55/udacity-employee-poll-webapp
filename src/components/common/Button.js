const Button = ({
  type,
  name,
  additionalClassNames = 'btn-primary',
  isDisabled,
  dataTestId = '',
}) => {
  return (
    <button
      type={type}
      className={`btn ${additionalClassNames}`}
      disabled={isDisabled}
      data-testid={`defaultButton${dataTestId}`}>
      {name}
    </button>
  );
};

export default Button;
