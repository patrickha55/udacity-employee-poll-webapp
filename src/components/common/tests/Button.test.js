import { render, screen } from '@testing-library/react';
import Button from '../Button';
import renderer from 'react-test-renderer';

describe('Test for Button component', () => {
  it('should renders snapshot', () => {
    const component = renderer.create(
      <Button
        name='test'
        type='button'
        className='btn-primary'
        disabled={false}
      />
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should renders the button component', () => {
    render(
      <Button
        name='test'
        type='button'
        className='btn-primary'
        disabled={false}
      />
    );

    const button = screen.getByTestId('defaultButton');
    const buttonName = screen.getByText('test');

    expect(button).toBeInTheDocument();
    expect(buttonName).toBeInTheDocument();
  });
});