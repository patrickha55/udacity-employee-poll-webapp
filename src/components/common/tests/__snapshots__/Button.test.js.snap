// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Button component should renders snapshot 1`] = `
<button
  className="btn btn-primary"
  data-testid="defaultButton"
  type="button"
>
  test
</button>
`;
